### PR TITLE
Field value and recording of ChecBoxChecked

### PR DESCRIPTION
**Pull request by janid1967**
abstract FieldValue class. Childs: CheckBoxChecked, Attachment and NumberValue
abstract RecordCommand/Validator and RecordCheckBoxCheckedCommand/Validator

RecordCheckBoxCheckedCommandHandler

Endpoint for recording number will come

### CODE REVIEW

File: src/Equinor.Procosys.Preservation.Command/TagCommands/RecordCommands/RecordCheckBoxChecked/RecordCheckBoxCheckedCommandHandler.cs [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+
+        public async Task<Result<Unit>> Handle(RecordCheckBoxCheckedCommand request, CancellationToken cancellationToken)
+        {
+            var tag = await _projectRepository.GetTagByTagIdAsync(request.TagId);
+
+            var reqDef = await _requirementTypeRepository.GetRequirementDefinitionByFieldIdAsync(request.FieldId);
+            var field = reqDef.Fields.Single(f => f.Id == request.FieldId);
+            var requirement = tag.Requirements.Single(r => r.RequirementDefinitionId == reqDef.Id);
+
+            var period = requirement.ActivePeriod;
```
Comment (hbjorgo) [inline src/Equinor.Procosys.Preservation.Command/TagCommands/RecordCommands/RecordCheckBoxChecked/RecordCheckBoxCheckedCommandHandler.cs]: Move all domain login into the domain.

File: src/Equinor.Procosys.Preservation.Domain/AggregateModels/ProjectAggregate/FieldValue.cs [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+        {
+            if (field == null)
+            {
+                throw new ArgumentNullException(nameof(field));
+            }
+            FieldId = field.Id;
+        }
+        
+        public int FieldId { get; private set; }
+        public string FieldType { get; set; }
```
Comment (hbjorgo) [inline src/Equinor.Procosys.Preservation.Domain/AggregateModels/ProjectAggregate/FieldValue.cs]: Set FieldType in subclass constructor?

Comment (janid1967) [reply src/Equinor.Procosys.Preservation.Domain/AggregateModels/ProjectAggregate/FieldValue.cs]: We can discuss this one

**Comment by janid1967** [comment]
@hbjorgo I think this one should be merged now.
More unit tests and validations on NA and N/A for number needed, but I want approval for business logic before starting that work

File: src/Equinor.Procosys.Preservation.Domain/AggregateModels/ProjectAggregate/FieldValue.cs (hunk 2 above)
Comment (janid1967) [reply src/Equinor.Procosys.Preservation.Domain/AggregateModels/ProjectAggregate/FieldValue.cs]: Removed FieldType. Changed HasDiscriminator to use softstrings which will avoid changes in DB if renaming classes

**Pull request merged**